File: LMProject/src/main/java/fpt/aptech/LMProject/entities/Category.java
package fpt.aptech.LMProject.entities;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;

@Entity
@Table(name = "Category")
@Data
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CategoryID")
    private int CategoryID;

    @Column(name = "CategoryName")
    private String  CategoryName;

 
    @JsonIgnore
    @OneToMany(mappedBy = "category")
    private List<Book> books;

}

File: backend/src/main/kotlin/com/example/kotlindemo/model/Category.kt
package com.example.kotlindemo.model

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table


@Entity
@Table(name = "category")
class Category {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  var categoryid: Int? = null
  var categoryname: String? = null // getters and setters
}
